daemon/logger/copier_test.go

```diff
@@ -205,6 +223,37 @@ func TestCopierSlow(t *testing.T) {
 	}
 }
 
+func TestCopierWithSized(t *testing.T) {
+	var jsonBuf bytes.Buffer
+	sizedLogger := &TestSizedLoggerJSON{Encoder: json.NewEncoder(&jsonBuf)}
+
+	logbuf := bytes.NewBufferString(strings.Repeat(".", sizedLogger.BufSize()*2))
+	rd, wr := io.Pipe()
+	c := NewCopier(map[string]io.Reader{"stdout": rd}, sizedLogger)
+	c.Run()
+	go func() {
+		io.Copy(wr, logbuf)
+		c.Close()
+	}()
+
```

Comment:
This needs a copier.Wait, I think?

Reply:
Yes it certainly does, one sec.

Reply:
This change was left out of the update I pushed, my apologies. I've updated the code now with the Wait to ensure the copy jobs are finished.